Don't double dispatch on CUDA logical_not.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38275 Don't double dispatch on CUDA logical_not.**
* #38274 Move Cuda Abs kernel to its own file.

This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

gchanan committed:
Don't double dispatch on UnarySignKernels.

This op doesn't need to be fast and the double dispatch is time consuming to compile.

[ghstack-poisoned]

gchanan commented:
I think(?) this has the same behavior and the cases I tried have the same behavior, but it would be good to confirm.

mruberry reviewed (commented):
The logical_not test https://github.com/pytorch/pytorch/blob/a553935e3c49c50cb6751a0714bd344f712c5f67/test/test_torch.py#L6399 is pretty milquetoast and doesn't use our latest @dtypes gadgetry or compare against NumPy. Want to take this opportunity to update it?

gchanan committed:
Update on "Don't double dispatch on CUDA logical_not."


This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

[ghstack-poisoned]

gchanan committed:
Update on "Don't double dispatch on CUDA logical_not."


This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

[ghstack-poisoned]

gchanan committed:
Update on "Don't double dispatch on CUDA logical_not."


This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

[ghstack-poisoned]

gchanan committed:
Update on "Don't double dispatch on CUDA logical_not."


This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

[ghstack-poisoned]

gchanan committed:
Update on "Don't double dispatch on CUDA logical_not."


This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

[ghstack-poisoned]

gchanan committed:
Update on "Don't double dispatch on CUDA logical_not."


This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

[ghstack-poisoned]

gchanan committed:
Update on "Don't double dispatch on CUDA logical_not."


This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

[ghstack-poisoned]

gchanan committed:
Update on "Don't double dispatch on CUDA logical_not."


This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

[ghstack-poisoned]

gchanan committed:
Update on "Don't double dispatch on CUDA logical_not."


This op doesn't need to be fast and the double dispatch is time consuming to compile.

Differential Revision: [D21511832](https://our.internmc.facebook.com/intern/diff/D21511832)

[ghstack-poisoned]
